### DISCUSSION TITLE
transcribe_audio.py crashes instantly upon start

### DISCUSSION THREAD

**Discussion by Imdrekaiser** [cyberofficial/Synthalingua#30] [Q&A] [answered]
I just downloaded all the things, ffmpeg, python 3.11.something, and ran the set up. 

Whenever I try to run anything the python script instantly fails. Doesn't matter whether it is the setup, transcribe_audio.py, or the example program, nothing runs. Opening the live translation example worked, until I changed it to try to find my other audio settings because it was recording my mic not my speakers.

Additionally the setup doesn't let me reinstall.

Trying to run the program via powershell or cmd causes an error
I did 'python transcribe_audio.py --list_microphones'
'C:\Users\(name)\AppData\Local\Programs\Python\Python311\python.exe: can't open file 'C:\\Users\\(name)\\transcribe_audio.py': [Errno 2] No such file or directory'
Trying to use the exact directory of the program is a syntax error.

Pc is using:
NVIDIA GeForce GTX 1060 3GB
AMD Ryzen 3 1200 Quad-Core Processor
24GB of RAM

So there should be no issue here

Please help.
Also, if this was answered before I didn't see it.

**Comment by cyberofficial** [comment]
The supported python environment is [Python 3.10.9](https://www.python.org/downloads/release/python-3109/), also the script is set in a way where you need to be in the current directory of the script.

**Reply by Imdrekaiser** [reply]
What do you mean by "the script is set in a way where you need to be in the current directory of the script."?

Also I'll test it with Python3.10.9 and edit the response later if I figured it out

**Comment by cyberofficial** [comment]
> What do you mean by "the script is set in a way where you need to be in the current directory of the script."?

When you edit the live translation script, it checks the current directory for the python environment. 
It will call for the python environment in the current folder where the script is ran.

<img width="910" alt="GitHubDesktop_dIoJaYMXDR" src="https://github.com/cyberofficial/Synthalingua/assets/19499442/2df061a2-52af-429d-b0af-9459f1027f88">

Make sure you edit the live translation script for the command flags and not directly run the python script.

**Reply by Imdrekaiser** [reply, marked as answer]
Was this the proper way to do it? Just edit the livetranslation file and run everything from that?
![image](https://github.com/cyberofficial/Synthalingua/assets/108836750/9263f498-6ee7-4510-bb90-29a150f42462)

**Reply by cyberofficial** [reply]
Yea, that is the correct way, since you have pretty long microphone names, best to use the index number when setting the microphone.

After you choose your right device, just use for ex `--set_microphone 17` and then add the other flags from [usage](https://github.com/cyberofficial/Synthalingua#usage).

**Reply by Imdrekaiser** [reply]
Alright, thank you very much. Now all I've got to do is find what microphone works, or follow your other discussion where you showed a guy how to use VoiceMeeter. I got it to work using Stereo Mix, but the TL had no idea what was going on and it cut out constantly.

An actual problem has raised though, whenever it transcribes a word/character it just puts a box with a question mark inside of it, I don't know how to fix that.

**Reply by cyberofficial** [reply]
the box with question mark means your operating system doesn't have additional languages installed. Though it shouldn't affected the actual program. see here https://www.reddit.com/r/Windows10/comments/tq5kkx/question_mark_in_boxes_appear_in_powershell_and/

**Reply by Imdrekaiser** [reply]
I've got JP, is that not it? 
![image](https://github.com/cyberofficial/Synthalingua/assets/108836750/9b948a33-ad7f-4e16-810f-6ec162957724)
you know what, this issue doesn't matter and I'll figure it out myself tomorrow.

But what I do want to know is if you have settings optimized for Okayu somewhere, or if I must tinker with settings.

**Comment by cyberofficial** [comment]
> I've got JP, is that not it? ![image](https://user-images.githubusercontent.com/108836750/250001995-9b948a33-ad7f-4e16-810f-6ec162957724.png) you know what, this issue doesn't matter and I'll figure it out myself tomorrow.
> 
> But what I do want to know is if you have settings optimized for Okayu somewhere, or if I must tinker with settings.

could just be windums being dumb per usual.
You can use the `--use_finetune` flag for optimized speech (trained on majority of the hololive speech patterns), the fine tune model can detect the members based on their voice printing of normal voice. Though Miko can still fry the program so be warned 🤣.
